src/extension.ts

```diff
@@ -0,0 +1,151 @@
+    }
+  }
+
+  console.log(`Location after path variables subsitution: ${executablePath}`);
+
+  // Check if the path points to a valid waspls executable
+  const executableStatus = await checkExecutable(executablePath);
+
+  if (executableStatus !== 'available') {
+    if (executableStatus === 'timedout') {
+      window.showInformationMessage(`The wasp server process has timed out.`);
+    } else {
+      if (userDefinedExecutablePath === '') {
+        window.showErrorMessage('No `waspls` executable is available in the VSCode PATH.');
+        return;
+      } else {
+        window.showInformationMessage(`The user defined executable path couldn't be run: [${executablePath}].`);
+      }
+    }
+  }
+
+
+  // TODO: send these settings to waspls so it can update its logging output if
+  // these are changed while the language server is running
+  const useOutputPanel = config.server.useOutputPanelForLogging;
+  const logFile = useOutputPanel ? "[OUTPUT]" : config.server.logFile;
+  const logFileOpt = logFile.trim() === '' ? [] : ['--log=' + logFile];
+
+  // Configure vscode-languageclient
+  let runArgs = [...logFileOpt];
+  let debugArgs = [...logFileOpt];
+
+  let serverOptions: ServerOptions = {
+    run: {
+      command: executablePath,
+      transport: TransportKind.stdio,
+      args: runArgs,
+    },
+    debug: {
+      command: executablePath,
+      transport: TransportKind.stdio,
+      args: debugArgs,
+    }
+  };
+
+  let clientOptions: LanguageClientOptions = {
+    documentSelector: [{ scheme: 'file', language: 'wasp' }],
+    initializationOptions: {
+      'vscode-wasp': workspace.getConfiguration('vscode-wasp'),
+    },
+    outputChannel: outputChannel
+  };
+
+  client = new LanguageClient(
+    'vscode-wasp',
+    'VSCode Wasp',
+    serverOptions,
+    clientOptions,
+  );
+
+  client.start();
+  outputChannel.appendLine('..Wasp LSP Server has been started..');
+
+  // Register command to restart waspls
+  context.subscriptions.push(commands.registerCommand('vscode-wasp.restartLanguageServer', async () => {
+    if (client) {
+      try {
+        console.log(`..Restarting Wasp LSP Server..`);
+        await client.restart();
+        console.log(`..Wasp LSP Server has been restarted..`);
+      } catch (err) {
+        outputChannel.appendLine(`Couldn't restart Wasp LSP Server: ${err.message}`);
+      }
+    }
+  }));
+}
+
+export function deactivate() {
+  if (client) {
+    client.stop();
+  }
+}
+
+// Runs the executable in `version` mode and checks that it:
+// - Exits successfuly
+// - Exits within 1 second
+//
+// If it fails either of these checks, a status is returned to show that the
+// executable path is not valid.
+async function checkExecutable(executablePath: string): Promise<Status> {
+  const execPromise = promisify(execFile)
+    (executablePath, ['version'], { timeout: 2000, windowsHide: true })
```

Comment:
Why 2s here?

Reply:
It's an arbitrary timeout. I picked it because 2s seemed like a reasonable amount of time to wait before deciding the executable isn't working.